File: app/src/standalone/java/io/github/hrilab/pepper_realtime/MovementController.java
package io.github.hrilab.pepper_realtime;

import android.util.Log;

/**
 * Stub implementation of MovementController for standalone mode (no robot hardware).
 * Logs all movement commands and simulates successful execution.
 */
public class MovementController {
    private static final String TAG = "MovementController[STUB]";
    
    public interface MovementListener {
        void onMovementStarted();
        void onMovementFinished(boolean success, String error);
    }
    
    private MovementListener listener;
    
    public void setListener(MovementListener listener) {
        this.listener = listener;
    }
    
    /**
     * Simulates moving Pepper in a specified direction
     */
    public void movePepper(Object qiContext, double distanceForward, double distanceSideways, double speed) {
        Log.i(TAG, String.format("🤖 [SIMULATED] Move - Forward: %.2fm, Sideways: %.2fm, Speed: %.2fm/s", 
            distanceForward, distanceSideways, speed));
        
        if (listener != null) {
            listener.onMovementStarted();
            // Simulate successful movement after a short delay
            new android.os.Handler(android.os.Looper.getMainLooper()).postDelayed(() -> {
                if (listener != null) {
                    listener.onMovementFinished(true, null);
                }
            }, 500);
        }
    }
    
    /**
     * Simulates turning Pepper
     */
    public void turnPepper(Object qiContext, double angleDegrees, double speed) {
        Log.i(TAG, String.format("🤖 [SIMULATED] Turn - Angle: %.1f°, Speed: %.2f", angleDegrees, speed));
        
        if (listener != null) {
            listener.onMovementStarted();
            new android.os.Handler(android.os.Looper.getMainLooper()).postDelayed(() -> {
                if (listener != null) {
                    listener.onMovementFinished(true, null);
                }
            }, 500);
        }
    }
    
    /**
     * Simulates looking at a position
     */
    public void lookAtPosition(Object qiContext, double x, double y, double z) {
        Log.i(TAG, String.format("🤖 [SIMULATED] Look at position - X: %.2f, Y: %.2f, Z: %.2f", x, y, z));
        
        if (listener != null) {
            listener.onMovementStarted();
            new android.os.Handler(android.os.Looper.getMainLooper()).postDelayed(() -> {
                if (listener != null) {
                    listener.onMovementFinished(true, null);
                }
            }, 300);
        }
    }
    
    /**
     * Cancels current movement (no-op in stub)
     */
    public void cancelCurrentMovement() {
        Log.i(TAG, "🤖 [SIMULATED] Cancel movement");
        if (listener != null) {
            listener.onMovementFinished(false, "Cancelled");
        }
    }
    
    /**
     * Shuts down the controller (no-op in stub)
     */
    public void shutdown() {
        Log.i(TAG, "🤖 [SIMULATED] Shutdown");
    }
}

File: app/src/standalone/java/io/github/anonymous/pepper_realtime/controller/MovementController.kt
package io.github.anonymous.pepper_realtime.controller

import android.os.Handler
import android.os.Looper
import android.util.Log

/**
 * Stub implementation of MovementController for standalone mode (no robot hardware).
 * Logs all movement commands and simulates successful execution.
 */
class MovementController {
    
    interface MovementListener {
        fun onMovementStarted()
        fun onMovementFinished(success: Boolean, error: String?)
    }

    companion object {
        private const val TAG = "MovementController[STUB]"
    }

    private var listener: MovementListener? = null

    fun setListener(listener: MovementListener?) {
        this.listener = listener
    }

    /**
     * Simulates moving Pepper in a specified direction
     */
    @Suppress("UNUSED_PARAMETER")
    fun movePepper(qiContext: Any?, distanceForward: Double, distanceSideways: Double, speed: Double) {
        Log.i(TAG, "🤖 [SIMULATED] Move - Forward: %.2fm, Sideways: %.2fm, Speed: %.2fm/s".format(
            distanceForward, distanceSideways, speed))

        listener?.let { l ->
            l.onMovementStarted()
            // Simulate successful movement after a short delay
            Handler(Looper.getMainLooper()).postDelayed({
                l.onMovementFinished(true, null)
            }, 500)
        }
    }

    /**
     * Simulates turning Pepper
     */
    @Suppress("UNUSED_PARAMETER")
    fun turnPepper(qiContext: Any?, direction: String, degrees: Double, speed: Double) {
        Log.i(TAG, "🤖 [SIMULATED] Turn - Direction: $direction, Angle: %.1f°, Speed: %.2f".format(degrees, speed))

        listener?.let { l ->
            l.onMovementStarted()
            Handler(Looper.getMainLooper()).postDelayed({
                l.onMovementFinished(true, null)
            }, 500)
        }
    }

    /**
     * Simulates looking at a position
     */
    @Suppress("UNUSED_PARAMETER")
    fun lookAtPosition(qiContext: Any?, x: Double, y: Double, z: Double) {
        Log.i(TAG, "🤖 [SIMULATED] Look at position - X: %.2f, Y: %.2f, Z: %.2f".format(x, y, z))

        listener?.let { l ->
            l.onMovementStarted()
            Handler(Looper.getMainLooper()).postDelayed({
                l.onMovementFinished(true, null)
            }, 300)
        }
    }

    /**
     * Simulates navigation to a saved location
     */
    @Suppress("UNUSED_PARAMETER")
    fun navigateToLocation(qiContext: Any?, savedLocation: Any?, speed: Float) {
        Log.i(TAG, "🤖 [SIMULATED] Navigate to location with speed: $speed m/s")

        listener?.let { l ->
            l.onMovementStarted()
            Handler(Looper.getMainLooper()).postDelayed({
                l.onMovementFinished(true, null)
            }, 1000)
        }
    }

    /**
     * Cancels current movement (no-op in stub)
     */
    fun cancelCurrentMovement() {
        Log.i(TAG, "🤖 [SIMULATED] Cancel movement")
        listener?.onMovementFinished(false, "Cancelled")
    }

    /**
     * Shuts down the controller (no-op in stub)
     */
    fun shutdown() {
        Log.i(TAG, "🤖 [SIMULATED] Shutdown")
    }
}

